client/src/FileManagerLayout.js

```diff
@@ -0,0 +1,120 @@
+import { useContext } from 'react';
+
+import {
+  DefaultErrorMessage,
+  Overlay,
+  OverlayContainer,
+  Spinner,
+  FullscreenButton
+} from 'flight-webapp-components';
+
+import {
+  Context as FileManagerContext,
+} from './FileManagerContext';
+
+import FileToolbar from './FIleToolbar';
+
+function FileManagerLayout({
+  children
+}) {
+  const { state, currentAbsDir, user, currentDir } = useContext(FileManagerContext);
+  let loadingMessage = null;
+  if (state !== 'connected' && state !== 'failed') {
+    loadingMessage = <Loading text="Loading file manager..." />;
+  }
+  let errorMessage = null;
+  if (state === 'failed') {
+    errorMessage = <DefaultErrorMessage />;
+  }
+
+  return (
+      <div className="overflow-auto">
+        <div className="row no-gutters">
+          <div className="col">
+            <div className="card-header toolbar text-light">
+              <div className="row no-gutters">
+                <div className="col">
+                  <div className="d-flex align-items-center">
+                    <h5 className="flex-grow-1 mb-0">
+                      <span className="font-weight-bolder">{user} - </span>
+                      {currentDir}
+                    </h5>
+                    <Toolbar
+                      fileManagerState={state}
+                      onZenChange={ () => {} }
+                      currentAbsDir={currentAbsDir}
+                    />
+                  </div>
+                </div>
+              </div>
+            </div>
+            <FileToolbar />
+            <div className="bg-white">
+              {loadingMessage}
+              {errorMessage}
+              {children}
+            </div>
+          </div>
+        </div>
+      </div>
+  );
+}
+
+function Toolbar({
+    onFullscreenChange,
+    onZenChange,
+    currentAbsDir,
+    fileManagerState
+}) {
+  const RefreshButton = fileManagerState === 'connected' ? (
+    <i
+      className="fa fa-arrows-rotate ml-2 link white-text"
+      title="refresh (Ctrl+R)"
+      onClick={refresh}
+    >
+    </i>
+  ) : null;
+
+  const ToConsoleButton = fileManagerState === 'connected' ? (
+    <i
+      title="Open in terminal"
+    >
+      <a
+        className="fa fa-terminal ml-2 link white-text"
```

Comment:
The margins are a bit uneven between the icons. In Console, the middle icon has `ml-2 mr-3`

Reply:
there is no mr-3 in my local Console branch but I've added mr-3 to this one.